numba/targets/arraymath.py

```diff
@@ -668,6 +668,101 @@ def _trivial_isnan(x):
         return _trivial_isnan
 
 
+@overload(np.iscomplex)
+def np_iscomplex(x):
+    def impl(x):
+        ax = np.asarray(x) # NumPy uses asanyarray here!
+        return ax.imag != 0
+    return impl
+
+
+@overload(np.isreal)
+def np_isreal(x):
+    def impl(x):
+        ax = np.iscomplex(x)
```

Comment:
This should not return False for types that cannot `type_can_asarray`, i.e. `np.iscomplex('string')`.

Reply:
On python2, I am having the following error when trying to use 'strings' as an input to a function:

```
Traceback (most recent call last):
  File "/path/to/numba/numba/tests/test_np_functions.py", line 733, in test_is_real_or_complex
    got = cfunc(x)
TypeError: can't unbox str type
```
Can python2 handle strings?

Reply:
It cannot. Unicode string support is Python 3 only.

Reply:
With the drop of python2 support, this issue that I was having with strings is solve.d